app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt

```diff
@@ -0,0 +1,66 @@
+package com.wire.android.navigation
+
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.runtime.Composable
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NamedNavArgument
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavType
+import androidx.navigation.navArgument
+import com.wire.android.ui.home.HomeScreen
+import com.wire.android.ui.settings.SettingsScreen
+import com.wire.android.ui.support.SupportScreen
+import com.wire.android.ui.home.userprofile.UserProfileScreen
+
+@ExperimentalMaterial3Api
+open class NavigationItem(
+    open val route: String,
+    val arguments: List<NamedNavArgument> = emptyList(),
+    open val content: @Composable (NavBackStackEntry) -> Unit
+    //TODO add animations here
+) {
+
+//    object Splash  //TODO
+//    object Login  //TODO
+
+    object Home : NavigationItem(
+        route = "home/{$HOME_START_TAB_ARGUMENT}",
```

Comment:
This is referencing the constant value "start_tab_index". Is this intended or shouldn't we change it dynamically with the route component of the home screen we are at?

Reply:
this `route` is used for creating the NavGraph, there we should pass the param name that we expect. 
And the actual param is passed while navigating (`navController.navigate()`), that's why I added `fun navigationRoute()`, it should be used for `navController.navigate()`